src/py21cmfast/src/IonisationBox.c

```diff
@@ -1688,12 +1688,30 @@ LOG_SUPER_DEBUG("freed fftw boxes");
 
     }
 
+    if (prev_redshift < 1){
+        free(previous_ionize_box->z_re_box);
+        if (flag_options->USE_MASS_DEPENDENT_ZETA) {
+            if (flag_options->USE_MINI_HALOS){
+                free(previous_ionize_box->Gamma12_box);
+                free(previous_ionize_box->dNrec_box);
+                free(previous_ionize_box->Fcoll);
+                free(previous_ionize_box->Fcoll_MINI);
+            }
+        }
+    }
+
     if(!flag_options->USE_TS_FLUCT) {
         if(user_params->USE_INTERPOLATION_TABLES) {
             freeSigmaMInterpTable();
         }
     }
 
+    if(INIT_ERFC_INTERPOLATION) {
```

Comment:
No, this shouldn't be done like this (well, at least not currently).

These arrays are generated once per session, thus they'll need to be freed differently. At the top of the file (line 194), where these are first allocated, instead do:
if(!flag_options->USE_MASS_DEPENDENT_ZETA && INIT_ERFC_INTERPOLATION) {

These are only used for the old, T_VIR dependence

Reply:
Can I confirm that you are saying to add `!flag_options->USE_MASS_DEPENDENT_ZETA` to both this line and 194 rather than saying freeing these two arrays here is wrong? I agree this is only used when USE_MASS_DEPENDENT_ZETA==False

Reply:
Ah, yeah, sorry for not being clear. I was saying to add `!flag_options->USE_MASS_DEPENDENT_ZETA` to line 194 (as you have done). However, you cannot free `ERFC_VALS` or `ERFC_VALS_DIFF`. Thus, I would remove the freeing of these. These are populated once per session, thus freeing them will cause segmentation faults.

We'll need to come up with a better way to free these. However, adding `!flag_options->USE_MASS_DEPENDENT_ZETA` to line 194 will at least ensure this isn't populated in most cases.

Reply:
Ah, I see what you mean. I didn't realize `INIT_ERFC_INTERPOLATION` is a global parameter rather than an input parameter. I will remove the freeing then